components/chef-run/lib/chef-run/cli_options.rb

```diff
@@ -0,0 +1,146 @@
+#
+# Copyright:: Copyright (c) 2018 Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef-run/text"
+require "chef-run/action/install_chef"
+
+# Moving the options into here so the cli.rb file is smaller and easier to read
+# For options that need to be merged back into the global ChefRun::Config object
+#   we do that with a proc in the option itself. We decided to do that because it is
+#   an easy, straight forward way to merge those options when they do not directly
+#   map back to keys in the Config global. IE, we cannot just do
+#   `ChefRun::Config.merge!(options)` because the keys do not line up, and we do
+#   not want all CLI params merged back into the global config object.
+# We know that the config is already loaded from the file (or program defaults)
+#   because the `Startup` class was invoked to start the program.
+module ChefRun
+  module CLIOptions
+
+    T = ChefRun::Text.cli
+    TS = ChefRun::Text.status
+
+    def self.included(klass)
+      klass.banner T.description + "\n" + T.usage_full
+
+      klass.option :version,
+        short: "-v",
+        long: "--version",
+        description:  T.version.description,
+        boolean: true
+
+      klass.option :help,
+        short: "-h",
+        long: "--help",
+        description:   T.help.description,
+        boolean: true
+
+      # Special note:
+      # config_path is pre-processed in startup.rb, and is shown here only
+      # for purpoess of rendering help text.
+      klass.option :config_path,
+        short: "-c PATH",
+        long: "--config PATH",
+        description: T.default_config_location(ChefRun::Config.default_location),
+        default: ChefRun::Config.default_location,
+        proc: Proc.new { |path| ChefRun::Config.custom_location(path) }
+
+      klass.option :identity_file,
+        long: "--identity-file PATH",
+        short: "-i PATH",
+        description: T.identity_file,
+        proc: (Proc.new do |paths|
+          path = paths
+          unless File.exist?(path)
+            raise OptionValidationError.new("CHEFVAL001", self, path)
+          end
+          path
+        end)
+
+      klass.option :ssl,
+        long: "--[no-]ssl",
+        description:  T.ssl.desc(ChefRun::Config.connection.winrm.ssl),
+        boolean: true,
+        default: ChefRun::Config.connection.winrm.ssl,
+        proc: Proc.new { |val| ChefRun::Config.connection.winrm.ssl(val) }
```

Comment:
Here you can see an example of the pattern for writing values back to the global config. See the comment at the top of this module and make sure the explanation there makes sense.